awx/ui_next/src/screens/Organization/shared/OrganizationForm.jsx

```diff
@@ -1,206 +1,174 @@
 
 import { OrganizationsAPI } from '@api';
-import { Config } from '@contexts/Config';
+import { ConfigContext } from '@contexts/Config';
+import AnsibleSelect from '@components/AnsibleSelect';
+import ContentError from '@components/ContentError';
+import ContentLoading from '@components/ContentLoading';
 import FormRow from '@components/FormRow';
 import FormField from '@components/FormField';
 import FormActionGroup from '@components/FormActionGroup/FormActionGroup';
-import AnsibleSelect from '@components/AnsibleSelect';
 import { InstanceGroupsLookup } from '@components/Lookup/';
+import { getAddedAndRemoved } from '@util/lists';
 import { required, minMaxValue } from '@util/validators';
 
-class OrganizationForm extends Component {
-  constructor(props) {
-    super(props);
-
-    this.getRelatedInstanceGroups = this.getRelatedInstanceGroups.bind(this);
-    this.handleInstanceGroupsChange = this.handleInstanceGroupsChange.bind(
-      this
-    );
-    this.handleSubmit = this.handleSubmit.bind(this);
-
-    this.state = {
-      instanceGroups: [],
-      initialInstanceGroups: [],
-      formIsValid: true,
-    };
-  }
-
-  async componentDidMount() {
-    let instanceGroups = [];
-
-    if (!this.isEditingNewOrganization()) {
-      try {
-        instanceGroups = await this.getRelatedInstanceGroups();
-      } catch (err) {
-        this.setState({ error: err });
-      }
-    }
-
-    this.setState({
-      instanceGroups,
-      initialInstanceGroups: [...instanceGroups],
-    });
-  }
-
-  async getRelatedInstanceGroups() {
-    const {
-      organization: { id },
-    } = this.props;
-    const { data } = await OrganizationsAPI.readInstanceGroups(id);
-    return data.results;
-  }
-
-  isEditingNewOrganization() {
-    const { organization } = this.props;
-    return !organization.id;
-  }
-
-  handleInstanceGroupsChange(instanceGroups) {
-    this.setState({ instanceGroups });
-  }
-
-  handleSubmit(values) {
-    const { handleSubmit } = this.props;
-    const { instanceGroups, initialInstanceGroups } = this.state;
-
-    const initialIds = initialInstanceGroups.map(ig => ig.id);
-    const updatedIds = instanceGroups.map(ig => ig.id);
-    const groupsToAssociate = [...updatedIds].filter(
-      x => !initialIds.includes(x)
+function OrganizationForm({ organization, i18n, me, onCancel, onSubmit }) {
+  const defaultVenv = {
+    label: i18n._(t`Use Default Ansible Environment`),
+    value: '/venv/ansible/',
+    key: 'default',
+  };
+  const { custom_virtualenvs } = useContext(ConfigContext);
+  const [contentError, setContentError] = useState(null);
+  const [hasContentLoading, setHasContentLoading] = useState(true);
+  const [initialInstanceGroups, setInitialInstanceGroups] = useState([]);
+  const [instanceGroups, setInstanceGroups] = useState([]);
+
+  const handleCancel = () => {
+    onCancel();
+  };
+
+  const handleSubmit = values => {
+    const { added, removed } = getAddedAndRemoved(
```

Comment:
This util `getAddedAndRemoved` cleaned up `handleSubmit` significantly. This is my first time seeing it :) Thanks for including that in our utils @keithjgrant